http/src/main/java/io/micronaut/http/HttpRequest.java

```diff
@@ -445,4 +445,24 @@ static <T> MutableHttpRequest<T> create(HttpMethod httpMethod, String uri, Strin
         Objects.requireNonNull(httpMethodName, "Argument [httpMethodName] is required");
         return HttpRequestFactory.INSTANCE.create(httpMethod, uri, httpMethodName);
     }
+
+    /**
+     * Returns a mutable request based on this request.
+     * @return the mutable request
+     * @since 4.7
+     */
+    default MutableHttpRequest<B> toMutableRequest() {
+        if (this instanceof MutableHttpRequest<B> mutableHttpRequest) {
```

Comment:
just override the method in MutableHttpRequest instead of using this instanceof

Reply:
I did also override it